From f4d05969295f9a4dcfe7ffce5fab4b446d926e2d Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Mon, 8 Jul 2019 16:06:43 +0200
Subject: [PATCH] TRITON-1783 VM creation for bhyve and kvm VMs shouldn't pass
 `quota` attribute to `vmadm`. Just `disks` and `flexible_disk_size` Reviewed
 by: Mike Gerdts <mike.gerdts@joyent.com> Approved by: Mike Gerdts
 <mike.gerdts@joyent.com>

---
 lib/workflows/provision.js | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/lib/workflows/provision.js b/lib/workflows/provision.js
index 8a53444..3125346 100644
--- a/lib/workflows/provision.js
+++ b/lib/workflows/provision.js
@@ -5,11 +5,12 @@
  */
 
 /*
- * Copyright (c) 2019, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
- * A brief overview of this source file: what is its purpose.
+ * This is the VM provision workflow, used to create provision jobs
+ * for Triton VMs.
  */
 
 var async = require('async');
@@ -23,7 +24,7 @@ var nfsVolumes = require('./nfs-volumes');
 
 var wfapiUrl;
 
-var VERSION = '8.2.1';
+var VERSION = '8.2.2';
 
 
 /*
@@ -231,6 +232,10 @@ function preparePayload(job, cb) {
                 payload[field] = job.params.image[field];
             }
         });
+
+        // Rely into default vmadm values with `disks` and `flexible_disk_size`
+        // for KVM/Bhyve VMs:
+        delete payload.quota;
     } else {
         payload['image_uuid'] = params['image_uuid'];
 
-- 
2.21.0

